### DISCUSSION TITLE
Requesting Help regarding implementation of Global Filter along with Controlled Pagination

### DISCUSSION THREAD

**Discussion by bhaskor-sarmah** [TanStack/table#2768] [Q&A] [answered]
Hi All,

I am a beginner and new to this, as a result, I am facing some issues. I tried to merge **Global Filter along with controlled pagination**. The Global Filter is working fine now, but when I try to change the page the Table gets refresh back to Page 0. Please help me learn how to do it correctly.

The code sandbox link is below
https://codesandbox.io/s/globalfilter-with-controlled-pagination-c55e5?from-embed

**Comment by bhaskor-sarmah** [comment, marked as answer]
I think the issue is resolved. I have taken the reference of the following [link](https://github.com/tannerlinsley/react-table/blob/master/docs/src/pages/docs/faq.md#how-can-i-manually-control-the-table-state). The solution I found is as follows (Don't know if its the correct way to do it):

`useTable(
    {
      columns,
      data,
      manualPagination: true,
      pageCount: controlledPageCount,
      autoResetPage: false,
      autoResetExpanded: false,
      autoResetGroupBy: false,
      autoResetSelectedRows: false,
      autoResetSortBy: false,
      autoResetFilters: false,
      autoResetRowState: false
    },
    useGlobalFilter,
    useSortBy,
    usePagination
  );`

**Reply by Davidcreador** [reply]
The part that worked for me was to set the `autoResetPage: false,` to false, that way the table doesn't reset. Thanks for this.
